user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserQuotas.java

```diff
@@ -0,0 +1,229 @@
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            if (diff != null && diff.size() > 0) {
+                zkClient.writeData("/config/users/" + username, updateUserJson(data, quotas));
+            }
+        } else {
+            log.debug("Creating quotas for user {}", username);
+            ensurePath("/config/users");
+            zkClient.createPersistent("/config/users/" + username, createUserJson(quotas));
+        }
+
+        notifyChanges(username);
+    }
+
+    /**
+     * Generates the JSON with the credentials
+     *
+     * @param quotas  quotas
+     * @return  Returns the generated JSON as byte array
+     */
+    protected byte[] createUserJson(JsonObject quotas)   {
+        JsonObject json = new JsonObject()
+                .put("version", 1)
+                .put("config", new JsonObject());
+
+        for (Map.Entry<String, Object> quota: quotas.getMap().entrySet()) {
+            json.getJsonObject("config").put(quota.getKey(), quota.getValue().toString());
+        }
+
+        return json.encode().getBytes(StandardCharsets.UTF_8);
+    }
+
+    /**
+     * Updates the quotas in existing JSON
+     *
+     * @param user JSON string with existing user configuration as byte[]
+     * @param quotas  quotas
+     *
+     * @return  Returns the updated JSON as byte array
+     */
+    protected byte[] updateUserJson(byte[] user, JsonObject quotas)   {
+        JsonObject json = new JsonObject(new String(user, StandardCharsets.UTF_8));
+
+        validateJsonVersion(json);
+
+        if (json.getJsonObject("config") == null)   {
+            json.put("config", new JsonObject());
+        }
+
+        for (Map.Entry<String, Object> quota: quotas.getMap().entrySet()) {
+            json.getJsonObject("config").put(quota.getKey(), quota.getValue().toString());
+        }
+        return json.encode().getBytes(StandardCharsets.UTF_8);
+
+    }
+
+    /**
+     * This notifies Kafka about the changes we have made
+     *
+     * @param username  Name of the user whose configuration changed
+     */
+    private void notifyChanges(String username) {
+        log.debug("Notifying changes for user {}", username);
+
+        ensurePath("/config/changes");
+
+        JsonObject json = new JsonObject().put("version", 2).put("entity_path", "users/" + username);
+        zkClient.createPersistentSequential("/config/changes/config_change_", json.encode().getBytes(StandardCharsets.UTF_8));
+    }
+
+    /**
+     * Ensures that the path in Zookeeper exists.
+     * It checks whether it already exists and in case it doesn't, it will create the path.
+     *
+     * @param path The Zookeeper path which should exist
+     */
+    private void ensurePath(String path)    {
+        if (!zkClient.exists(path))   {
+            zkClient.createPersistent(path, true);
+        }
+    }
+
+    /* test */
+    public boolean isPathExist(String path)    {
```

Comment:
Does this need to be public? If it's non-private for testing purposes then why can't it be package access?